### ISSUE TITLE
#452 query translation compilation issue

### ISSUE THREAD

**Issue by roboLoki** [SOCI/soci#688]
Hi-

I think #452 is still hanging around (kind of). I notice the creators like clang and ms compiler. I use mingw with cygwin. At first I thought I did something spacey but then it looks like my code is bombing out because it doesn't like auto_ptr. I looked at a change log and saw where some lines were taken out (conditional compiling). I am not certain that works for my environment. Not sure what to do. It is a big deal because the query transformations let you build a query as you go like qbe piecing them together. Any insight is appreciated.

thx. roboLoki

 I will leave you with my warnings and error msg:

mister-201@LAPTOP-HT7A13B8 ~/swigwin-3.0.12/proj1
$ x86_64-w64-mingw32-c++.exe -c -std=c++11 sociTest.cpp -I"C:\cygwin64\home\mister-201\soci\include\soci" -I"C:\cygwin64\home\mister-201\soci\include" -I"C:\Program Files\Firebird\Firebird_2_5\include" -I"C:\cygwin64\lib\gcc\x86_64-w64-mingw32\6.4.0\include" -I"C:\cygwin64\home\mister-201\swigwin-3.0.12\proj1" -I"C:\ActiveTcl\include" -L"C:\cygwin64\home\mister-201\swigwin-3.0.12\proj1" -Lc:/ActiveTcl/lib
In file included from soci.h:33:0,
                 from sociTest.cpp:15:
/home/mister-201/soci/include/soci/rowset.h:151:24: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
     const cxx_details::auto_ptr<statement> st_;
                        ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-w64-mingw32/6.4.0/include/c++/memory:81:0,
                 from sociTest.cpp:1:
/usr/lib/gcc/x86_64-w64-mingw32/6.4.0/include/c++/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
In file included from soci.h:33:0,
                 from sociTest.cpp:15:
/home/mister-201/soci/include/soci/rowset.h:152:24: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
     const cxx_details::auto_ptr<T> define_;
                        ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-w64-mingw32/6.4.0/include/c++/memory:81:0,
                 from sociTest.cpp:1:
/usr/lib/gcc/x86_64-w64-mingw32/6.4.0/include/c++/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
In file included from soci.h:34:0,
                 from sociTest.cpp:15:
/home/mister-201/soci/include/soci/session.h:46:49: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
     void set_query_transformation_(cxx_details::auto_ptr<details::query_transformation_function>& qtf);
                                                 ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-w64-mingw32/6.4.0/include/c++/memory:81:0,
                 from sociTest.cpp:1:
/usr/lib/gcc/x86_64-w64-mingw32/6.4.0/include/c++/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
In file included from soci.h:34:0,
                 from sociTest.cpp:15:
/home/mister-201/soci/include/soci/session.h: In member function ‘void soci::session::set_query_transformation(T)’:
/home/mister-201/soci/include/soci/session.h:86:22: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
         cxx_details::auto_ptr<details::query_transformation_function> qtf(new details::query_transformation<T>(callback));
                      ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-w64-mingw32/6.4.0/include/c++/memory:81:0,
                 from sociTest.cpp:1:
/usr/lib/gcc/x86_64-w64-mingw32/6.4.0/include/c++/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
mister-201@LAPTOP-HT7A13B8 ~/swigwin-3.0.12/proj1
$ x86_64-w64-mingw32-c++.exe  sociTest.o tcl86t.lib tk86t.lib libsoci_core_4_0.a  libsoci_firebird_4_0.a libsoci_empty_4_0.a fbclient.dll ib_util.dll  -o sociTest.exe                  sociTest.o:sociTest.cpp:(.text$_ZN4soci7session24set_query_transformationI5orderEEvT_[_ZN4soci7session24set_query_transformationI5orderEEvT_]+0x65): undefined reference to `soci::session::set_query_transformation_(std::auto_ptr<soci::details::query_transformation_function>&)'
collect2: error: ld returned 1 exit status

**Comment by mloskot** [comment]
SOCI is C++03 library, nit C++11 or 14.

FYI, we test against GCC and clang on Linux and MSVC on Windows.

**Comment by mloskot** [comment]
FYI, we have MinGW build job on AppVeyor https://ci.appveyor.com/project/SOCI/soci/branch/master

**Comment by roboLoki** [comment]
no. i tried it. i am trying to find my install script options to retry
installing without the c++ 11 options for cmake. i had them all in a file.
i will reinstall with different cmake options. thx

**Comment by roboLoki** [comment]
hi mateusz-
i rebuilt soci without turning on c++ compatibility . i got a query
transformation to work. good enough for now. of note is i linked with 3 dll
files this time. thx. for sticking with me. must be something i did wrong.
thx again - gavin
